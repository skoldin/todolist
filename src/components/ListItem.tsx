import React from "react";

interface Props {
  item: Item;
  updateItem: (item: Item) => void;
  deleteItem: (id: string) => void;
}

const ListItem: React.FunctionComponent<Props> = ({
  item,
  updateItem,
  deleteItem
}) => {
  const { name, done, id } = item;

  const handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void = e => {
    const { checked } = e.currentTarget;

    item.done = checked;

    updateItem(item);
  };

  const handleDelete = () => {
    deleteItem(item.id);
  };

  return (
    <li className="list-group-item d-flex justify-content-between align-items-center">
      <div className="custom-control custom-checkbox">
        <input
          type="checkbox"
          className="custom-control-input"
          id={id}
          defaultChecked={done}
          onChange={handleChange}
        />
        <label
          className="custom-control-label"
          htmlFor={id}
          style={done ? { color: "#999", textDecoration: "line-through" } : {}}
        >
          {name}
        </label>
      </div>
      <button className="btn btn-link" onClick={handleDelete}>
        Delete
      </button>
    </li>
  );
};

export default ListItem;
