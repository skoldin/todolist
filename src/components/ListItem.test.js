import React from "react";
import ListItem from "./ListItem";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const item = { id: "", name: "Test item", done: true };
const props = {
  item,
  updateItem: jest.fn(),
  deleteItem: jest.fn()
};

let Component = renderer.create(<ListItem {...props} />);
let wrapper;

beforeEach(() => {
  wrapper = shallow(<ListItem {...props} />);
});

it("render matches the snapshot", () => {
  expect(Component.toJSON()).toMatchSnapshot();
});

it("handle change", () => {
  const event = { currentTarget: { checked: false } };

  const input = wrapper.find("input");

  input.simulate("change", event);

  expect(props.updateItem).toHaveBeenCalledWith(props.item);
});

it("handle delete", () => {
  const button = wrapper.find("button");

  button.simulate("click");

  expect(props.deleteItem).toHaveBeenCalledWith(props.item.id);
});
