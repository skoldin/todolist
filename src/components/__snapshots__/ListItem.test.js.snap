// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches the snapshot 1`] = `
<li
  className="list-group-item d-flex justify-content-between align-items-center"
>
  <div
    className="custom-control custom-checkbox"
  >
    <input
      className="custom-control-input"
      defaultChecked={true}
      id=""
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="custom-control-label"
      htmlFor=""
      style={
        Object {
          "color": "#999",
          "textDecoration": "line-through",
        }
      }
    >
      Test item
    </label>
  </div>
  <button
    className="btn btn-link"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
